### PR TITLE
feat: reorganize repo for Aurora builds

### PR DESCRIPTION
**Pull request by m2Giles**
Reorganization of repo in order to support Aurora.

Containerfile restructured to remove inline shell. Instead everything is built out of sourced shell files in `/tmp/build`

This is still a work in progress. The github actions need to be updated.

### CODE REVIEW

File: build_files/base/build.sh [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#!/usr/bin/bash
+
+set -oue pipefail
+
+. /tmp/build/copr-repos.sh
```
[the remark this answers is not in the document]
Comment (m2Giles) [reply build_files/base/build.sh:5]: That could also work. With this method, we can inline things like that singular rpm-ostree install. But I like the loop method to make us more purposeful with changes. I personally like this source method to not worry about the +x bit.

File: Containerfile [hunk 2]
```diff
@@ -15,114 +15,27 @@ ARG IMAGE_FLAVOR="${IMAGE_FLAVOR}"
...
-    rm -f /etc/yum.repos.d/charm.repo && \
-    rm -f /etc/yum.repos.d/ublue-os-staging-fedora-"${FEDORA_MAJOR_VERSION}".repo && \
-    echo "Hidden=true" >> /usr/share/applications/fish.desktop && \
-    echo "Hidden=true" >> /usr/share/applications/htop.desktop && \
-    echo "Hidden=true" >> /usr/share/applications/nvtop.desktop && \
-    echo "Hidden=true" >> /usr/share/applications/gnome-system-monitor.desktop && \
-    rm -f /etc/yum.repos.d/_copr_che-nerd-fonts-"${FEDORA_MAJOR_VERSION}".repo && \
-    sed -i '/^PRETTY_NAME/s/Silverblue/Bluefin/' /usr/lib/os-release && \
+# Build, cleanup, commit.
+RUN bash -c ". /tmp/build/build.sh"  && \
```
[the remark this answers is not in the document]
Comment (m2Giles) [reply Containerfile:34]: That's why I'm using the source method. It doesn't matter if the file has the execute bit or not.

File: Containerfile [hunk 3]
```diff
@@ -15,114 +15,27 @@ ARG IMAGE_FLAVOR="${IMAGE_FLAVOR}"
...
-  install -c -m 0755 /tmp/starship /usr/bin && \
-  echo 'eval "$(starship init bash)"' >> /etc/bashrc
 
+# Copy ublue-update.toml to tmp first, to avoid being overwritten.
+COPY /system_files/shared/usr/etc/ublue-update/ublue-update.toml /tmp/ublue-update.toml
 # Copy Bluefin CLI packages
 COPY --from=ghcr.io/ublue-os/bluefin-cli /usr/bin/atuin /usr/bin/atuin
 COPY --from=ghcr.io/ublue-os/bluefin-cli /usr/share/bash-prexec /usr/share/bash-prexec
+# COPY ublue kmods, add needed negativo17 repo and then immediately disable due to incompatibility with RPMFusion
+COPY --from=ghcr.io/ublue-os/akmods:${AKMODS_FLAVOR}-${FEDORA_MAJOR_VERSION} /rpms /tmp/akmods-rpms
```
[the remark this answers is not in the document]
Comment (m2Giles) [reply Containerfile:31]: That would be nicer. We could remove a few of these COPY layers with that.

File: Containerfile (hunk 2 above)
[the remark this answers is not in the document]
Comment (p5) [reply Containerfile]: Ah, I didn't realise that's how the sourcing works!  SGTM

**Review by castrojo — APPROVED** [review]

**Pull request merged**